File: LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginActivityModel.java
package com.charlye934.loginddagger.login;

public class LoginActivityModel implements LoginActivityMVP.Model{

    private LoginRepository repository;

    public LoginActivityModel(LoginRepository repository){
        this.repository = repository;
    }

    @Override
    public void createUser(String firstName, String lastName) {
        //Logica de busines
        repository.saveUser(new User(firstName,lastName));
    }

    @Override
    public User getUser() {
        //Logica de busines
        return repository.getUser();
    }
}

File: RxJavaDagger/LoginDagger/app/src/main/java/com/charlye934/loginddagger/login/LoginActivityModel.kt
package com.charlye934.loginddagger.login

class LoginActivityModel(private val repository:LoginRepository) : LoginActivityMVP.Model{

    override fun createUser(firstName: String, lastName: String) {
        repository.saveUser(User(firstName,lastName))
    }

    override fun getUser(): User = repository.getUser()
}